flake-module.nix

```diff
@@ -73,6 +73,13 @@ in
                 default = hp: { };
                 defaultText = ''Build tools useful for Haskell development are included by default.'';
               };
+              checks = mkOption {
+                type = types.attrsOf (types.nullOr types.bool);
```

Comment:
This is a weird type for a `checks` attribute, but after some reading, the real problem is that it's over-engineered.
I'd suggest an option `enableHLSCheck` with `types.bool` and a good description.